pySDC/implementations/problem_classes/AdvectionEquation_ND_FD.py

```diff
@@ -114,25 +114,42 @@ def __init__(
         )
         self.A *= -c
 
-        self.xv = np.meshgrid(*[xvalues for _ in range(ndim)])
+        self.xvalues = xvalues
         self.Id = sp.eye(np.prod(nvars), format='csc')
 
         # store relevant attributes
-        self.c, self.freq, self.sigma, self.dx = c, freq, sigma, dx
-        self.stencil_type, self.order = stencil_type, order
-        self.lintol, self.liniter = lintol, liniter
-        self.direct_solver, self.bc = direct_solver, bc
+        self.freq, self.sigma = freq, sigma
+        self.lintol, self.liniter, self.direct_solve = lintol, liniter, direct_solver
+
+        # read-only attributes
+        self._readOnly = [ndim, c, stencil_type, order]
```

Comment:
I like this, but I would prefer a dictionary here. Since later you access variables from this list, I think it's more straightforward if you can look for a string key, not just an index. But this is fewer lines, so I don't insist. :)

Reply:
Just did that. But black absolutely want to put those properties in two lines, which add unnecessary lines. @pancetta can we set black to allow one-line functions when it's just a return statement ?